dll/win32/shell32/wine/pidl.c

```diff
@@ -2030,11 +2032,9 @@ DWORD _ILGetDrive(LPCITEMIDLIST pidl, LPWSTR pOut, UINT uSize)
  */
 BOOL _ILIsUnicode(LPCITEMIDLIST pidl)
 {
-    LPPIDLDATA lpPData = _ILGetDataPointer(pidl);
-
     TRACE("(%p)\n",pidl);
 
-    return (pidl && lpPData && PT_VALUEW == lpPData->type);
+    return (_ILGetFSType(pidl) & PT_FS_UNICODE_FLAG) != 0;
```

Comment:
Shouldn't these changes have some `#ifndef __REACTOS__ ...` too?
Or are they imported from upstream?

Reply:
We are not syncing shell32 with Wine. The correct implementation for Wine would be closer to `PT_VALUEW == lpPData->type || PT_FOLDERW == lpPData->type` but even that is not correct for things on the desktop.

Reply:
Yeah, I asked because the other change does have it. Anyway.

Reply:
Yes, the other does have it just in case someone decides to compare with wine and cherry pick things that might seem correct on first glance.

Ideally we should delete _ILIsUnicode and just let the debug dumper code figure it out on its own. No other code should be asking. If you want the filename, ask IShellFolder/IShellItem.